### PR TITLE
Extend golang docs: additional dependencies and tooling semantics

### PR DESCRIPTION
**Pull request by lubgr**
Minor improvements to the docs on `golang` hooks:

- In `golang` hooks, `additional_dependencies` can be used. According to the existing docs on other language hooks, I have added a note on that. See also issue #797.
- The second paragraph is a summary of my findings from setting up `golang` hooks. It might be worth adding this for the benefit of others?

### CODE REVIEW

**Comment by asottile** [comment]
I would remove the second paragraph.  there are many poorly written tools in every language so it is not necessary to call them out in the golang section

**Comment by lubgr** [comment]
Thanks for having a look so quickly!

The second paragraph was not commenting on UI of Go tools, but pointing out a fundamental mismatch of how the entire Go ecosystem works (including the compiler), where individual files have no meaning, only packages contained in directories have. The tools that refuse to work on individual files are hence fully streamlined and provide the least surprise for Go devs. As a consequence, it's hard to write Go hooks for Go code that work as nicely as hooks written in other languages.

But thinking about it again, this only applies to Go hooks written to work with _Go code_, not Go hooks generally, for any other language.

So long story short, I have remove the paragraph 🙂

**Review by asottile — APPROVED** [review]
